packages/oid4vci-demo-frontend/src/router/sequencer.ts

```diff
@@ -0,0 +1,117 @@
+            throw new Error('The sequence element in the ecosystem json is missing steps')
+        }
+
+        this.stepsById = this.sequenceConfig.steps.reduce((map, step) => {
+            map[step.id] = step
+            return map
+        }, {} as { [key: string]: VCIConfigSequenceStep })
+    }
+
+    public getDefaultRoute(state?: any): string {
+        // Take the first navigate step in the list with isDefaultRoute set to true
+        for (const step of this.sequenceConfig.steps) {
+            switch (step.operation) {
+                case VCIOperation.NAVIGATE:
+                    if (step.isDefaultRoute) {
+                        const navStep = step as VCINavigationStep
+                        if (!navStep.path) {
+                            throw new Error(`Field path of navigation step with id ${navStep.id} is empty!`)
+                        }
+                        return navStep.path
+                    }
+            }
+        }
+        throw new Error('No navigation steps have been defined in the sequence element of the ecosystem json')
+    }
+
+    /*
+        We call setCurrentRoute on every page because
+        a. to get the useNavigate() hook and
+        b. because the user can go to that pages using the browser buttons or history and the sequencer needs to know which page the user is actually on
+    */
+    public setCurrentRoute(route: string, navigateFunction: NavigateFunction) {
+        this.navigateFunction = navigateFunction
+        const defaultRoute = this.getDefaultRoute()
+        for (const step of this.sequenceConfig.steps) {
+            switch (step.operation) {
+                case VCIOperation.NAVIGATE:
+                    const path = (step as VCINavigationStep).path
+                    if (path === route || (route === '/' && path == defaultRoute)) {
+                        this.currentStep = step
+                        return
+                    }
+                    break
+            }
+        }
+        throw new Error(`No step has been defined for the current route ${route} in the sequence element of the ecosystem json`)
+    }
+
+    public async next(state?: any) {
+        if (this.currentStep === undefined) {
+            throw new Error('current route/step is unknown')
+        }
+        if (this.currentStep.nextId) {
+            await this.goToStep(this.currentStep.nextId, state)
+        } else {
+            throw new Error(`There is no next step defined in step ${this.currentStep.id} in the sequence element of the ecosystem json`)
+        }
+    }
+
+    public async goToStep(stepId: string, state?: any) {
+        const nextStep = this.stepsById[stepId]
+        if (!nextStep) {
+            throw new Error(`Could not find a step id ${stepId} which was defined as nextId of step ${this.currentStep?.id}`)
+        }
+        switch (nextStep.operation) {
+            case VCIOperation.NAVIGATE:
```

Comment:
Rename VCIOperation to RouteOperation. Since we are now going into a dynamic route based solution, we are using VCI in some cases but also OID4VP/SIOPv2 in others. So then calling it VCIOperation is not really handy